pkg/api/middlewares.go

```diff
@@ -27,3 +29,52 @@ func getHTTPCacheGenNumberHeaderSetterMiddleware(cacheGenNumbersLoader *purger.T
 		})
 	})
 }
+
+// HTTPHeaderMiddleware adds specified HTTPHeaders to the request context
+type HTTPHeaderMiddleware struct {
+	TargetHeaders []string
+}
+
+// NewHttpHeaderMiddleware creates a new hTTPHeaderMiddleware given a string containing headers delimited by ~ (whitespaces in headers ignored)
+func NewHttpHeaderMiddleware(unparsedTargetHeaders string) HTTPHeaderMiddleware {
+	removeWhitespaceRegex, err := regexp.Compile(`\s`)
+	if err != nil {
+		util_log.Logger.Log("WhitespaceRegexCompilationStatus", "Failed")
+		return HTTPHeaderMiddleware{TargetHeaders: nil}
+	}
+	headersWithNoWhitespace := removeWhitespaceRegex.ReplaceAllString(unparsedTargetHeaders, "")
+
+	splitRegex, err2 := regexp.Compile(`~`)
+	if err2 != nil {
+		util_log.Logger.Log("SplitRegexCompilationStatus", "Failed")
+		return HTTPHeaderMiddleware{TargetHeaders: nil}
+	}
+	splitHeaders := splitRegex.Split(headersWithNoWhitespace, -1)
```

Comment:
I think we can remove all this if we figure it out how to use the flagext.StringSlice